### PR TITLE
[release-1.0.0-RC3] update sdk 1.9.0

### PR DESCRIPTION
**Pull request by stafyniaksacha**
update kuzzle-sdk to 1.9.0 (kuzzle 1.0.0-rc3)
* implement new dataCollectionFactory signature
* implement new roles management
* remove yolo fix

### CODE REVIEW

File: src/javascripts/authentication/authorizationApi.service.js [hunk 1]
```diff
@@ -2,24 +2,52 @@ require('./session.service');
...
         },
         canDeleteIndex: function (index) {
-          return true;
+          return this.canDoAction(index, '*', 'admin', 'deleteIndex');
         },
         canDoAction: function (index, collection, controller, action) {
-          return true;
+          // wait for lazy loading rights
+          if (!session.session.rights) {
+            return canDoAction(index, collection, controller, action);
```
Comment (xbill82) [inline src/javascripts/authentication/authorizationApi.service.js]: Doesn't this cause a blocking loop until the rights have been fetched?

**Comment by xbill82** [comment]
+1

**Comment by scottinet** [comment]
+1

**Comment by ballinette** [comment]
+1

**Pull request merged**